src/DataObject/DataObjectDocument.php

```diff
@@ -561,6 +561,19 @@ public function getDataObject(): DataObject
             );
         }
 
+        // Allow extensions to provide a fallback for retrieving the DataObject
+        // This is useful for modules that need special handling (e.g., Fluent with locale-specific content)
+        if (!$dataObject) {
```

Comment:
perhaps check if the above two changes mean that you don't need this one anymore?